eloquent.md

```diff
@@ -329,47 +332,84 @@ Your application may run out of memory if you attempt to load tens of thousands
 
 The `chunk` method will retrieve a subset of Eloquent models, passing them to a closure for processing. Since only the current chunk of Eloquent models is retrieved at a time, the `chunk` method will provide significantly reduced memory usage when working with a large number of models:
 
-    use App\Models\Flight;
+```php
+use App\Models\Flight;
 
-    Flight::chunk(200, function ($flights) {
-        foreach ($flights as $flight) {
-            //
-        }
-    });
+Flight::chunk(200, function ($flights) {
+    foreach ($flights as $flight) {
+        //
+    }
+});
+```
 
 The first argument passed to the `chunk` method is the number of records you wish to receive per "chunk". The closure passed as the second argument will be invoked for each chunk that is retrieved from the database. A database query will be executed to retrieve each chunk of records passed to the closure.
 
 If you are filtering the results of the `chunk` method based on a column that you will also be updating while iterating over the results, you should use the `chunkById` method. Using the `chunk` method in these scenarios could lead to unexpected and inconsistent results. Internally, the `chunkById` method will always retrieve models with an `id` column greater than the last model in the previous chunk:
 
-    Flight::where('departed', true)
-            ->chunkById(200, function ($flights) {
-                $flights->each->update(['departed' => false]);
-            }, $column = 'id');
+```php
+Flight::where('departed', true)
+    ->chunkById(200, function ($flights) {
+        $flights->each->update(['departed' => false]);
+    }, $column = 'id');
+```
+
+<a name="streaming-results-lazily"></a>
+### Streaming Results Lazily
+
+The `lazy()` method works similarly to [the `chunk` method](#chunking-results), executing the query in chunks. However, instead of passing each chunk into a callback, the `lazy()` method returns [a `LazyCollection`](/docs/{{version}}/collections#lazy-collections) of Eloquent models, which lets you interact with the results as a single stream:
+
+```php
+use App\Models\Flight;
+
+foreach (Flight::lazy() as $flight) {
+    //
+}
```

Comment:
@JosephSilber I'm wondering if `$flight` is an instance of Eloquent or a chunk set of Eloquent?

Reply:
> instead of passing each chunk into a callback, the `lazy()` method returns [...] results as a single stream

So `$flight` is a single Eloquent model.

Reply:
1. I thought that was clear enough. Is it not?

2. If it isn't, would this help?

    ![image](https://user-images.githubusercontent.com/1403741/112420355-6f7fd900-8d03-11eb-9aed-4c23050fd3ff.png)

Reply:
ability to pass `chunkCount` throws me off there. Understood now

Reply:
The `$chunkCount` determines how many records should be requested per chunk behind the scenes.

Bigger chunks use more memory, why smaller chunks means more queries (potentially exacerbated by eager loading relationships).